build.gradle.kts

```diff
@@ -0,0 +1,268 @@
+    "runtime" {
+        isVisible = false
+        extendsFrom(configurations["coreRuntime"])
+    }
+    "gradlePlugins" {
+        isVisible = false
+    }
+    "testRuntime" {
+        extendsFrom(configurations["runtime"])
+        extendsFrom(configurations["gradlePlugins"])
+    }
+
+    all {
+        attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage::class.java, Usage.JAVA_RUNTIME))
+    }
+}
+
+extra.set("allTestRuntimeDependencies", configurations["testRuntime"].allDependencies)
+
+val patchedExternalModulesDir = File(buildDir, "external/files")
+val patchedExternalModules = files(Callable { fileTree(patchedExternalModulesDir).files.sorted() })
+patchedExternalModules.builtBy("patchExternalModules")
+
+dependencies {
+    "externalModules"("org.gradle:gradle-kotlin-dsl:${BuildEnvironment.gradleKotlinDslVersion}")
+    "externalModules"("org.gradle:gradle-kotlin-dsl-tooling-builders:${BuildEnvironment.gradleKotlinDslVersion}")
+    "coreRuntime"(project(":launcher"))
+    "coreRuntime"(project(":runtimeApiInfo"))
+    "runtime"(project(":wrapper"))
+    "runtime"(project(":installationBeacon"))
+    "runtime"(patchedExternalModules)
+    pluginProjects.forEach { "gradlePlugins"(it) }
+    implementationPluginProjects.forEach { "gradlePlugins"(it) }
+    "gradlePlugins"(project(":workers"))
+    "gradlePlugins"(project(":dependencyManagement"))
+    "gradlePlugins"(project(":testKit"))
+
+    "coreRuntimeExtensions"(project(":dependencyManagement")) //See: DynamicModulesClassPathProvider.GRADLE_EXTENSION_MODULES
+    "coreRuntimeExtensions"(project(":pluginUse"))
+    "coreRuntimeExtensions"(project(":workers"))
+    "coreRuntimeExtensions"(patchedExternalModules)
+}
+
+extra.set("allCoreRuntimeExtensions", configurations["coreRuntimeExtensions"].allDependencies)
+
+tasks.create<PatchExternalModules>("patchExternalModules") {
+    allModules = configurations["externalModulesRuntime"]
+    coreModules = configurations["coreRuntime"]
+    modulesToPatch = configurations["externalModules"]
+    destination = patchedExternalModulesDir
+}
+
+evaluationDependsOn(":distributions")
+
+tasks.create<Install>("install") {
+    description = "Installs the minimal distribution into directory ${project.property("gradle_installPath")}"
+    group = "build"
+    with(project(":distributions").property("binDistImage") as CopySpec)
+    installDirPropertyName = "gradle_installPath"
+}
+
+tasks.create<Install>("installAll") {
+    description = "Installs the full distribution into directory ${project.property("gradle_installPath")}"
+    group = "build"
+    with(project(":distributions").property("allDistImage") as CopySpec)
+    installDirPropertyName = "gradle_installPath"
+}
+
+afterEvaluate {
+    if (gradle.startParameter.isBuildCacheEnabled()) {
+        rootProject.availableJavaInstallations.validateBuildCacheConfiguration((gradle as GradleInternal).settings.buildCache)
```

Comment:
:o: Seems weird that your own Gradle build has to use an internal API to achieve the logic you want.